sdcm/utils/remote_logger.py

```diff
@@ -302,6 +315,131 @@ def _logger_cmd(self) -> str:
         return f"{cmd} >> {self._target_log_file} 2>&1"  # pylint: disable=protected-access
 
 
+class K8sClientLogger(LoggerBase):  # pylint: disable=too-many-instance-attributes
+    READ_REQUEST_TIMEOUT = 300
+    RECONNECT_DELAY = 30
+    CHUNK_SIZE = 128
+
+    def __init__(self, pod: "sdcm.cluster_k8s.BasePodContainer", target_log_file: str):
+        """
+        Reads logs from a k8s pod using k8s client API and forwards it to a file.
+
+        It is self-healing and will re-open the stream if it is closed and continue reading logs from place it stopped.
+        """
+        super().__init__(target_log_file)
+        self._pod_name = pod.name
+        self._parent_cluster = pod.parent_cluster
+        self._last_log_timestamp = ""
+        self._last_read_timestamp = None
+        self._k8s_core_v1_api = KubernetesOps.core_v1_api(self._parent_cluster.k8s_cluster.get_api_client())
+        self._termination_event = ThreadEvent()
+        self._file_object = open(self._target_log_file, "a", encoding="utf-8")  # pylint: disable=consider-using-with
+        self._log_reader = None
+        self._thread = Thread(target=self._log_loop)
+
+    def start(self):
+        self._log.info("Starting logger for pod %s", self._pod_name)
+        self._thread.start()
+
+    def stop(self, timeout=None):
+        self._log.info("Stopping logger for pod %s", self._pod_name)
+        self._termination_event.set()
+        self._thread.join(timeout)
+        self._file_object.close()
+
+    @retrying(n=20, sleep_time=3, allowed_exceptions=(ConnectionError, ))
+    def _open_stream(self) -> None:
+        if self._termination_event.is_set():
+            return
+        self._log.debug("Opening log reader stream for pod %s", self._pod_name)
+        # read log since last read timestamp + 60 seconds margin
```

Comment:
How did you come to the 60 sec margin?

Reply:
just wanted to be some time before last read, but after thinking I don't think we need it (just a fraction of sec should be enough).